### PR TITLE
Sysbuild additions A

### PR DESCRIPTION
**Pull request by nordicjm**

### CODE REVIEW

File: sysbuild/CMakeLists.txt.orig [hunk 1]
```diff
@@ -0,0 +1,102 @@
+#
```
Comment (tejlmand) [inline sysbuild/CMakeLists.txt.orig:1]: what is a `CMakeLists.txt.orig` file doing in the PR ?

**Review by tejlmand — CHANGES_REQUESTED** [review]
Please remove the `.orig` file.

File: sysbuild/Kconfig.dfu [hunk 2]
```diff
@@ -0,0 +1,31 @@
+# Copyright (c) 2023 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+
+config DFU_MULTI_IMAGE_PACKAGE_BUILD
+	bool "Created multi-image DFU"
+	depends on BOOTLOADER_MCUBOOT
+	help
+	  I don't actually know what this does or what the point of this file is, maybe I should look that up one day.
```
Comment (tejlmand) [inline sysbuild/Kconfig.dfu:9]: 🤔  Think this should be rephrased before merging.

**Review by tejlmand — CHANGES_REQUESTED** [review]

**Review by tejlmand — APPROVED** [review]

File: sysbuild/CMakeLists.txt [hunk 3]
```diff
@@ -41,6 +41,114 @@ function(${SYSBUILD_CURRENT_MODULE_NAME}_pre_cmake)
...
+        set_target_properties(mcuboot PROPERTIES BOARD nrf9160dk_nrf9160@${BOARD_REVISION})
+      else()
+        set_target_properties(mcuboot PROPERTIES BOARD nrf9160dk_nrf9160)
+      endif()
+    elseif(NOT DEFINED mcuboot_BOARD AND "${BOARD}" STREQUAL "thingy91_nrf9160_ns")
+      # MCUboot must run in secure mode on the nRF9160
+      if(DEFINED BOARD_REVISION)
+        set_target_properties(mcuboot PROPERTIES BOARD thingy91_nrf9160@${BOARD_REVISION})
+      else()
+        set_target_properties(mcuboot PROPERTIES BOARD thingy91_nrf9160)
```
Comment (57300) [inline sysbuild/CMakeLists.txt:57]: I know this is temporary, but there has to be a more efficient way to tabulate this.

File: sysbuild/CMakeLists.txt [hunk 4]
```diff
@@ -71,22 +179,42 @@ endfunction(${SYSBUILD_CURRENT_MODULE_NAME}_post_cmake)
 # Consider if this shoulc come through Sysbuild Kconfig flag.
 set(NCS_SYSBUILD_PARTITION_MANAGER TRUE)
 
+list(APPEND CMAKE_MODULE_PATH ${ZEPHYR_NRF_MODULE_DIR}/cmake/sysbuild/modules)
+include(ncs_sysbuild_extensions)
```
Comment (57300) [inline sysbuild/CMakeLists.txt:183]: There's already `nrf/sysbuild/extensions.cmake`. Should those be combined?

File: sysbuild/netcore.cmake [hunk 5]
```diff
@@ -41,22 +41,45 @@ if(SB_CONFIG_NETCORE_802154_RPMSG)
...
     list(APPEND PM_DOMAINS ${SB_CONFIG_NETCORE_802154_RPMSG_DOMAIN})
   endif()
-  list(APPEND PM_${SB_CONFIG_NETCORE_802154_RPMSG_DOMAIN}_IMAGES
-      "${SB_CONFIG_NETCORE_802154_RPMSG_NAME}"
-  )
   set_property(GLOBAL APPEND PROPERTY
-      PM_${SB_CONFIG_NETCORE_802154_RPMSG_DOMAIN}_IMAGES
-      ${PM_${SB_CONFIG_NETCORE_802154_RPMSG_DOMAIN}_IMAGES}
+               PM_${SB_CONFIG_NETCORE_802154_RPMSG_DOMAIN}_IMAGES
+               "${SB_CONFIG_NETCORE_802154_RPMSG_NAME}"}
```
Comment (57300) [inline sysbuild/netcore.cmake:46]: Caused a syntax warning:
```suggestion
               "${SB_CONFIG_NETCORE_802154_RPMSG_NAME}"
```

File: sysbuild/secureboot.cmake [hunk 6]
```diff
@@ -32,9 +32,24 @@ if(SB_CONFIG_SECURE_BOOT)
...
+
+    if(SB_CONFIG_BOOTLOADER_MCUBOOT)
+      ExternalNcsVariantProject_Add(APPLICATION mcuboot VARIANT ${image})
+    else()
+      ExternalNcsVariantProject_Add(APPLICATION ${DEFAULT_IMAGE} VARIANT ${image})
+    endif()
+
+    set_property(GLOBAL APPEND PROPERTY
+        PM_${SB_CONFIG_SECURE_BOOT_DOMAIN}_IMAGES
+        "s1_image"
```
Comment (57300) [inline sysbuild/secureboot.cmake:51]: nit:
```suggestion
        "${image}"
```

**Review by 57300 — COMMENTED** [review]
I can't review the secure boot stuff.

File: sysbuild/CMakeLists.txt (hunk 4 above)
Comment (nordicjm) [reply sysbuild/CMakeLists.txt:183]: @tejlmand

File: sysbuild/CMakeLists.txt (hunk 3 above)
Comment (nordicjm) [reply sysbuild/CMakeLists.txt:57]: It needs to check board revision for each, so each board will have 2 flows, I don't see a way to reduce that

File: sysbuild/secureboot.cmake (hunk 6 above)
Comment (nordicjm) [reply sysbuild/secureboot.cmake]: Fixed

File: sysbuild/netcore.cmake (hunk 5 above)
Comment (nordicjm) [reply sysbuild/netcore.cmake]: Fixed

File: sysbuild/CMakeLists.txt (hunk 3 above)
Comment (57300) [reply sysbuild/CMakeLists.txt:57]: I was thinking of condensing it like this for a start:
```cmake
if(BOARD STREQUAL "nrf9160dk_nrf9160_ns")
  set(board_secure nrf9160dk_nrf9160)
elseif(BOARD STREQUAL "thingy91_nrf9160_ns")
  set(board_secure thingy91_nrf9160)
elseif(BOARD STREQUAL "nrf5340dk_nrf5340_cpuapp_ns")
  set(board_secure nrf5340dk_nrf5340_cpuapp)
elseif(BOARD STREQUAL "thingy53_nrf5340_cpuapp_ns")
  set(board_secure thingy53_nrf5340_cpuapp)
endif()

if(DEFINED board_secure AND DEFINED BOARD_REVISION)
  set(board_secure ${board_secure}@${BOARD_REVISION})
endif()

...

if(NOT DEFINED mcuboot_BOARD AND DEFINED board_secure)
  # MCUboot must run in secure mode
  set_target_properties(mcuboot PROPERTIES BOARD ${board_secure})
endif()

...

if(NOT DEFINED b0_BOARD AND DEFINED board_secure)
  # b0 must run in secure mode
  set_target_properties(b0 PROPERTIES BOARD ${board_secure})

  if(SB_CONFIG_BOOTLOADER_MCUBOOT)
    set_target_properties(s1_image PROPERTIES BOARD ${board_secure})
  endif()
endif()
```

Comment (nordicjm) [reply sysbuild/CMakeLists.txt]: Updated, thanks

**Pull request merged**